index.bs

```diff
@@ -1402,6 +1405,87 @@ lt="ReadableStreamDefaultReader(stream)">new ReadableStreamDefaultReader(<var>st
   1. Perform ! ReadableStreamReaderGenericRelease(*this*).
 </emu-alg>
 
+<h5 id="default-reader-iterator" method for="ReadableStreamDefaultReader">
+  iterator({ cancel = true } = {})
+</h5>
+
+<div class="note">
+  The <code>iterator</code> method returns an async iterator which can be used to consume the stream. The return method
+  of this iterator will optionally release or cancel this reader.
+</div>
+
+<emu-alg>
+  1. If ! IsReadableStreamDefaultReader(*this*) is *false*, throw a *TypeError* exception.
+  1. If *this*.[[ownerReadableStream]] is *undefined*, throw a *TypeError* exception.
+  1. Let _iterator_ be ? ObjectCreate(`<a idl>ReadableStreamDefaultReaderAsyncIteratorPrototype</a>`, « [[Reader]],
+  [[Cancel]] »).
+  1. Set _iterator_.[[Reader]] to *this*.
+  1. Set _iterator_.[[Cancel]] to ! ToBoolean(_cancel_).
+  1. Return _iterator_.
+</emu-alg>
+
+<h5 id="default-reader-@@asynciterator" for="ReadableStreamDefaultReader">[@@asyncIterator]()</h5>
+
+<div class="note">
+  The <code>@@asyncIterator</code> method is an alias of {{ReadableStreamDefaultReader/iterator}}.
+</div>
+
+<emu-alg>
+  1. Let _iterator_ be ? GetMethod(*this*, "iterator").
+  1. Return ? Call(_iterator_, *this*).
+</emu-alg>
+
+<h3 id="default-reader-asynciterator-prototype" interface lt="ReadableStreamDefaultReaderAsyncIteratorPrototype">
+  ReadableStreamDefaultReaderAsyncIteratorPrototype
+</h3>
```

Comment:
It should be written somewhere that this inherits from [%AsyncIteratorPrototype%](https://tc39.github.io/ecma262/#sec-asynciteratorprototype).

Reply:
yea ik... i wasn't sure how/where to do that.

Reply:
is this a must for implementers? Or is it how things are spec'ed?

Reply:
@mcollina all async iterators should inherit from (have [[Prototype]] set to) %AsyncIteratorPrototype%, but it doesn't happen by default. i added it in my last commit.

Reply:
@mcollina Yes; having it would allow

```js
for await (const chunk of stream.getReader().getIterator()) {
}
```

in addition to just

```js
for await (const chunk of stream.getReader()) {
}
```

This is fully consistent with how sync iterators operate, that you can do `Array.from(map.entries())` in addition to just `Array.from(map)`.

Reply:
Also, you can get the %AsyncIteratorPrototype% through `Object.getPrototypeOf(Object.getPrototypeOf((async function* () {}).prototype))`.